### PR TITLE
Mixed source compile test reproducing issues/707

### PR DESCRIPTION
**Pull request by restingbull**
* Integration test helpr RulesKotlinWorkspace

https://github.com/bazelbuild/rules_kotlin/issues/707

### CODE REVIEW

File: src/test/kotlin/io/bazel/kotlin/integration/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    name = "lib",
+    srcs = glob(
+        ["*.kt"],
+        exclude = ["*Test.kt"],
+    ),
+)
+
+kt_jvm_test(
+    name = "WriteWorkspaceTest",
+    size = "enormous",
```
Comment (cgruber) [inline src/test/kotlin/io/bazel/kotlin/integration/BUILD.bazel:13]: Heh

File: src/test/kotlin/io/bazel/kotlin/integration/MixedSourceCompileTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package io.bazel.kotlin.integration
+
+import com.google.common.truth.Truth
```
Comment (cgruber) [inline src/test/kotlin/io/bazel/kotlin/integration/MixedSourceCompileTest.kt:3]: Unnecessary?

File: src/test/kotlin/io/bazel/kotlin/integration/RulesKotlinWorkspace.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
+package io.bazel.kotlin.integration
+
+import io.bazel.kotlin.integration.WriteWorkspace.BzlWorkspace
+import io.bazel.kotlin.integration.WriteWorkspace.Workspace
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardOpenOption.CREATE_NEW
+import java.util.concurrent.TimeUnit
+
+class RulesKotlinWorkspace private constructor(
```
Comment (cgruber) [inline src/test/kotlin/io/bazel/kotlin/integration/RulesKotlinWorkspace.kt:10]: Can we start adding some class level kdocs, at least for new types please?

File: src/test/kotlin/io/bazel/kotlin/integration/WriteWorkspace.kt [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+import java.nio.file.Files
+import java.nio.file.Files.exists
+import java.nio.file.Path
+import java.nio.file.StandardOpenOption
+import java.nio.file.StandardOpenOption.CREATE_NEW
+import java.nio.file.attribute.FileTime
+import java.time.Instant
+import kotlin.streams.toList
+
+object WriteWorkspace {
```
Comment (cgruber) [inline src/test/kotlin/io/bazel/kotlin/integration/WriteWorkspace.kt:14]: The naming of this singleton is a wee bit weird - but I really like the dsl.

A few example usages in a kdoc here would be super handy, so someone isn't hunting down examples in the tests.

**Review by cgruber — APPROVED** [review]
Ok, this is pretty nice. I like how it sets the stage for better integration tests than the examples. Hearty approval. Just some missing inline docs, but it's more a nag than a blocker.

**Review by Bencodes — APPROVED** [review]

**Pull request merged**